### DISCUSSION TITLE
AttributeError: module 'whisper' has no attribute 'load_model'

### DISCUSSION THREAD

**Discussion by learngvrk** [openai/whisper#251] [Q&A] [answered]
Hi,

I am trying to use the whisper module within a container and as I am accessing the load_model attribute. How to resolve this issue. Kindly help.

    # Transcribe the Decoded Audio file
    model = whisper.load_model("base")
    result = model.transcribe(dest)
    print(result["text"])

AttributeError: module 'whisper' has no attribute 'load_model'

**Comment by egbakou** [comment, marked as answer]
Hi @learngvrk
Make sure you install whisper using the command `pip install git+https://github.com/openai/whisper.git `
Take a look at this similar question that was answered in the discussion: https://github.com/openai/whisper/discussions/143

**Reply by leslieetubo** [reply]
Tried but same issue

**Reply by leslieetubo** [reply]
Env is python 3.8.5

**Reply by leslieetubo** [reply]
I found my error. I named the python script whisper.py. Make sure you chose a different name else you may hit a circular import error.

**Reply by egbakou** [reply]
Marked this answered, so the OpenAI team will be focus on another questions.

**Reply by initialvisuals** [reply]
@Leslieetubo  Thank you that solved my issue, I also had my file named whisper.py, changed it to speechtotext.ph ;D

**Reply by aerdengiz** [reply]
I now have the same problem: I did download whisper from GitHub, it worked fine until 4 days ago. I'm running Jupyter notebook, Anaconda 2.3.2, MacBook Pro M1: 
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Input In [6], in <cell line: 1>()
----> 1 model = whisper.load_model('base')

AttributeError: module 'whisper' has no attribute 'load_model'

The only thing I changed is update Anaconda from v 2.3.1 to v.2.3.2!

**Reply by aerdengiz** [reply]
I fixed the problem though I'm not sure how and why. There was a problem with ipywidgets not being up to date and updating Jupyter notebooks eliminated the problem.

**Reply by harshrumbum** [reply]
my problem is this thank you

**Reply by Allakazan** [reply]
Doesn't freaking work. I'm running inside WSL and even installing from git gives the same error

**Comment by learngvrk** [comment]
Thank you for your response. Your suggestion did solve the problem.

Ranjith

**Comment by kapousa** [comment]
> install git+https://github.com/openai/whisper.git

I follow you advise to solve the issue but after installing whisper this command, I got this error:
`
ERROR: Could not find a version that satisfies the requirement torch (from openai-whisper) (from versions: none)
`
`
ERROR: No matching distribution found for torch
`

I'm using Apple MacBook Air M1

**Reply by leslieetubo** [reply]
@kapousa, preflight checklist:

- [x] Using rosetta stone to install python! important, 
- [x] Create a new virtual environment?
- [x] Used `pip install --no-cache-dir [package]` to not get the cached version.
- [x]  If all fails, try restarting. It helps 
I did the installation on a Macbook Pro M1, but if the error persists, try getting an older release. I have not encountered the error before, but I will try to replicate the error. Good Luck.

**Reply by tantan39** [reply]
Did you successfully run it? I'm getting the same issue

**Reply by leslieetubo** [reply]
Yes I did, run it. Also make sure the file you are trying to run is not called ```whisper.py``` name it something else.

**Comment by porfirionoel** [comment]
Hi team, Could someone tell me if they have implemented it in EC2 AWS in ubuntu, I get error "AttributeError: module 'whisper' has no attribute 'load_model' when trying to run it, I reinstall it and it keeps coming out, I hope someone can help me?

**Reply by egbakou** [reply]
Hi @porfirionoel, 
Take a look at this similar issue  https://github.com/openai/whisper/discussions/143

**Comment by santhosh-sp** [comment]
Hello, 
I trained a model using whisper fine-tuned in colab and saved model into local/colab. How do I load model using path?.

I used both 

1. whisper.Whisper.load_trained("/mode/path")  
2. whisper.load_model("/model./path")

None of them are working.

Please suggest how to load custom model?

Thanks.

**Comment by SeunghyunSEO** [comment]
i encountered this error too, 

```
(Pdb) whisper.load_model
*** AttributeError: module 'whisper' has no attribute 'load_model'
```

anybody fix this ?

**Reply by VR-13** [reply]
If you install whisper using pip install -U openai-whisper instead of pip install whisper, it might solve your error. It did for me anyways :)

**Reply by yasergsi** [reply]
This works

**Reply by augiot** [reply]
This method works

**Comment by kinsuky** [comment]
Hello, I get this error, I am using wisper in google colab, it says the following: ( module stable_whisper has no attribute results_to_sentence_srt ) what can I do? Could someone help me, I cannot generate the srt files

**Reply by charan-debug** [reply]
Same issue I'm facing in Collab, did you get the solution to solve it?

**Reply by lauriestark** [reply]
Same issue here in Colab, none of the above suggestions solved it for me.

**Reply by mahima5598** [reply]
For Colab,
1. Restart the session
2. If you have already installed --upgrade openai, do not install whisper again.

This worked for me. 

For knowledge related to AI subscribe to my newsletter: https://www.linkedin.com/build-relation/newsletter-follow?entityUrn=7165846835213914112

**Comment by alaryutin123** [comment]
If anyone still having an issue - I had a stupid noob mistake - I had a file 'whisper.py' in the same directory as the script I was trying to run that uses 'import whisper'. So it simply used this file as a module input and needless to say - crashed.

**Reply by rahaf-fahad** [reply]
lol this is so funny , when i read your comment, I laughed and said it was impossible, but I said I would check to make sure, and it turned out that I actually have a file with the same name, and this is the cause of the problem, lol. Thank you!
